aiopg/connection.py

```diff
@@ -449,7 +449,11 @@ def echo(self):
 
     if PY_341:  # pragma: no branch
         def __del__(self):
-            if not self._conn.closed:
+            try:
+                _conn = self._conn
+            except AttributeError:
+                return
+            if _conn is not None and not _conn.closed:
```

Comment:
I am not sure it is a good idea: the whole point of this `__del__` method is issuing a warning and calling `self.close()` if there was an issue. Making `close()` resitent to `self._conn` not being set will clutter that method for no reson IMHO